src/Microsoft.Health.Fhir.Core.UnitTests/Features/Operations/Export/ExportProcessingJobTests.cs

```diff
@@ -133,16 +133,79 @@ public async Task GivenAnExportJob_WhenItFinishesAPageOfResultsAndAFollowupJobEx
             Assert.Equal(followUpJob, queueClient.JobInfos[0]);
         }
 
-        private string GenerateJobRecord(OperationStatus status, string failureReason = null)
+        [Theory]
+        [InlineData("Patient", "Observation", null, null)]
+        [InlineData(null, null, "range1", "range2")]
+        public async Task GivenAnExportJob_WhenItFinishesAPageOfResultsAndNewerParallelJobExists_ThenANewProgressJobIsQueued(string resourceType1, string resourceType2, string feedRange1, string feedRange2)
+        {
+            string progressResult = string.Empty;
+
+            Progress<string> progress = new Progress<string>((result) =>
+            {
+                progressResult = result;
+            });
+
+            var queueClient = new TestQueueClient();
+            var processingJob = new ExportProcessingJob(MakeMockJobWithProgressUpdate, queueClient);
+
+            var runningJob = GenerateJobInfo(GenerateJobRecord(OperationStatus.Running, resourceType: resourceType1, feedRange: feedRange1));
+
+            var followUpJob = GenerateJobInfo(GenerateJobRecord(OperationStatus.Running, resourceType: resourceType2, feedRange: feedRange2));
+            followUpJob.Id = runningJob.Id + 1;
+            queueClient.JobInfos.Add(followUpJob);
+
+            var taskResult = await processingJob.ExecuteAsync(runningJob, progress, CancellationToken.None);
+
+            Assert.True(queueClient.JobInfos.Count == 2);
+            Assert.DoesNotContain(_progressToken, queueClient.JobInfos[0].Definition);
+            Assert.Equal(followUpJob, queueClient.JobInfos[0]);
+        }
+
+        [Fact]
+        public async Task GivenAnExportJob_WhenItFinishesAPageOfResultAndCanceledGroupJobInQueue_ThenANewProgressJobIsNotQueued()
+        {
+            string progressResult = string.Empty;
+
+            Progress<string> progress = new Progress<string>((result) =>
+            {
+                progressResult = result;
+            });
+
+            var queueClient = new TestQueueClient();
+            var processingJob = new ExportProcessingJob(MakeMockJobWithProgressUpdate, queueClient);
+
+            var runningJob = GenerateJobInfo(GenerateJobRecord(OperationStatus.Running));
+
+            var existingCanceledJob = GenerateJobInfo(GenerateJobRecord(OperationStatus.Running));
+            existingCanceledJob.Id = runningJob.Id - 2;
+            existingCanceledJob.Status = JobStatus.Cancelled;
+            queueClient.JobInfos.Add(existingCanceledJob);
+
+            var existingCanceledPendingJob = GenerateJobInfo(GenerateJobRecord(OperationStatus.Running));
+            existingCanceledPendingJob.Id = runningJob.Id - 1;
+            existingCanceledPendingJob.CancelRequested = true;
+            queueClient.JobInfos.Add(existingCanceledPendingJob);
```

Comment:
Should these be two different theories instead of both being part of one test? IE: one for cancelled and one for cancel requested?